internal/services/forwarder/service_test.go

```diff
@@ -59,3 +69,155 @@ func TestRender(t *testing.T) {
 		assert.Equal(t, expected, actual)
 	})
 }
+
+func TestService(t *testing.T) {
+	t.Run("stop", func(t *testing.T) {
+		p := gomonkey.ApplyFunc(tele.NewBot, func(_ tele.Settings) (*tele.Bot, error) {
+			return &tele.Bot{Me: &tele.User{Username: "mock"}}, nil
+		})
+		defer p.Reset()
+		_, _, s := setup(t, "hookId", "confirmationString", "token", 123)
+		errCh := make(chan error)
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() { errCh <- s.Run(ctx) }()
+		cancel()
+
+		assert.NoError(t, <-errCh)
+	})
+	t.Run("error get interaction", func(t *testing.T) {
+		p := gomonkey.ApplyFunc(tele.NewBot, func(_ tele.Settings) (*tele.Bot, error) {
+			return &tele.Bot{Me: &tele.User{Username: "mock"}}, nil
+		})
+		defer p.Reset()
+		q, buf, s := setup(t, "hookId", "confirmationString", "token", 123)
+		ctx, cancel := context.WithCancel(context.Background())
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() { defer wg.Done(); _ = s.Run(ctx) }()
+
+		q.Put(entities.Message{
+			InteractionId: "",
+			Type:          entities.MessageTypeNew,
+			Text:          "Hello",
+			VkSenderId:    1234,
+		})
+
+		cancel()
+		wg.Wait()
+
+		logOutput := buf.String()
+		assert.Contains(t, logOutput, `get interaction failed`)
+		assert.Contains(t, logOutput, `interactionId=""`)
+	})
+	t.Run("empty interaction", func(t *testing.T) {
```

Comment:
Wasn't the previous test about empty interaction? Maybe the test names in this test and the previous test are swapped by mistake

Reply:
until the implementation is not clear I consider nil err and nil interaction a valid case